packages/filter-workspace-packages/src/getChangedPackages.ts

```diff
@@ -18,10 +18,16 @@ export default async function changedSince (packageDirs: string[], commit: strin
       changedPkgs.push(packageDir)
     }
   }
-  return changedPkgs
+
+  const ignoreDependentForPkgs = Array.from(dirsMatchingFilter)
+    .map(dir => path.join(repoRoot, dir))
+    .map(dir => changedPkgs.find(pkg => dir.startsWith(pkg)))
+    .filter(dir => !!dir)
+
+  return [changedPkgs, ignoreDependentForPkgs as string[]]
```

Comment:
maybe instead of returning two arrays, we could return one array with objects

Something like `Array<{ dir: string, changeType: test | source }>`

Reply:
I'll still have to combine them later into a list of a specific type, since it's used to find the changes packages. Do you still want to do it?

Reply:
it seems to me that it will require fewer loops and/or searches in the array

Reply:
sure, done [here](https://github.com/pnpm/pnpm/pull/3035/commits/4ca66d65645a36c0143ec9b1c6cb75c5e32d866c)